microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/commons/CodeActionData.java

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.commons;
+
+import java.util.Objects;
+
+/**
+ * Represents the data that all code actions have.
+ */
+public class CodeActionData {
+
+	public static enum CodeActionId {
```

Comment:
The main problem with enum is that it i snot extensable. As some extension can contribute to code action, the STring id should be used.

We have the similar usecase in lemminx (with diagnostic code). The main idea is that we provide an IXMLErrorCode API and we implement it with enum, see https://github.com/eclipse/lemminx/search?q=IXMLerrorcode

Reply:
Okay, I've implemented an extensible id system